vnext/ReactWindowsCore/WinRTWebSocketResource.h

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+
+#pragma once
+
+//#include <Mso/eventWaitHandle/eventWaitHandle.h>
+#include <winrt/Windows.Networking.Sockets.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <IWebSocketResource.h>
+
+#include <future>
+#include <queue>
+
+namespace Microsoft::React
+{
+
+class WinRTWebSocketResource : public IWebSocketResource
+{
+  winrt::Windows::Foundation::Uri m_uri;
+  winrt::Windows::Networking::Sockets::MessageWebSocket m_socket;
+  winrt::Windows::Networking::Sockets::MessageWebSocket::MessageReceived_revoker m_revoker;
+  winrt::Windows::Storage::Streams::DataWriter m_writer;
+
+  std::atomic_bool m_connectRequested;
+  std::atomic_bool m_sendRequested;
+  // Mso::ManualResetEvent m_connectPerformed;
+  // Mso::ManualResetEvent m_sendPerformed;
+  std::promise<void> m_connectPerformed;
+
+  CloseCode m_closeCode;
+  std::string m_closeReason;
+  std::queue<std::pair<std::string, bool>> m_writeQueue;
```

Comment:
I mentioned this somewhere else, but seeing the type of writeQueue here as a pair of string and bool can lead to confusion of what the types are for. It's preferable to use a struct here so we can get names.